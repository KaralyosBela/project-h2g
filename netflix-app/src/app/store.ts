import { configureStore } from "@reduxjs/toolkit";
import moviesReducer from "../features/moviesSlice";
import hehe from "../features/actions"

export const store = configureStore({
  reducer: {
    movies: moviesReducer,
    bruh: hehe
  }
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
