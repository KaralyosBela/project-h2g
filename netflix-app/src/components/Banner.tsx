import { useState } from "react";
import { AddMovieModal } from "./AddMovieModal";
import classes from "./Banner.module.css";
import { useDispatch } from "react-redux"
import { getMovies, addMovie } from "../features/moviesSlice"
import { AppDispatch } from "../app/store";

export const Banner: React.FC = () => {
  const [openAddModal, setOpenAddModal] = useState<boolean>(false);

  const dispatch = useDispatch<AppDispatch>();


  const hideModal = () => {
    setOpenAddModal(false);
  }

  const feccs = () => {
    dispatch(getMovies())
  }

  return (
    <div className={classes.picture}>
      <AddMovieModal show={openAddModal} hide={hideModal}/>
      <div className={classes.banner}>
        <div className={classes.upperBar}>
          <div>
            <h4 className={classes.upperBarTitle}>
              <span>netflix</span>Roulette
            </h4>
          </div>
          <button className={classes.addButton} onClick={()=> {setOpenAddModal(!openAddModal); dispatch(addMovie({
            title: "string",
            release_date: "string",
            genre: "string",
            thumbnail: "string",
            movie_url: "string",
            rating: "string",
            runtime: "string",
          }))}}>+ ADD MOVIE</button>
        </div>
        <div className={classes.title}>
          <h1>FIND YOUR MOVIE</h1>
        </div>
        <div className={classes.searchBar}>
          <input placeholder="What do you want to watch?"></input>
          <button className={classes.searchbutton} onClick={feccs}>SEARCH</button>
        </div>
      </div>
    </div>
  );
};
