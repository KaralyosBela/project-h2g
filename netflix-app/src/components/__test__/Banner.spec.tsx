import { fireEvent, render, screen } from "@testing-library/react";
import { Banner } from "../Banner";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import "@testing-library/jest-dom/extend-expect";

// const mockUseState = jest.fn();

jest.mock("../../components/AddMovieSuccessModal", () => ({
  AddMovieSuccessModal: () => <div>Add Movie Success Modal</div>,
})).mock("../../components/AddMovieModal", () => ({
  AddMovieModal: () => <div>Add Movie Modal</div>,
}));
// .mock("react", () => ({
//   useState: () => mockUseState(),
// }));

describe("Banner component", () => {

  it("should render the component without the modals", () => {
    const {container} = render(
      <Provider store={store}>
        <Banner />
      </Provider>
    );
  
    expect(container).toMatchSnapshot();
    // screen.debug();
  });

  
it("should render the component with the 'Add Modal'", () => {
  const {container} = render(
    <Provider store={store}>
      <Banner />
    </Provider>
  );

  const addButton = screen.getByText(/add movie/i);
  fireEvent.click(addButton);

  expect(container).toMatchSnapshot();
  // screen.debug();
});

it("should render the component with the 'Add Modal Success'", () => {
  const {container} = render(
    <Provider store={store}>
      <Banner />
    </Provider>

  );

  const addButton = screen.getByText(/add movie/i);
  fireEvent.click(addButton);
  //nem tudok hivatkozni komponensen beluli elemre
  // const submitButton = screen.getByText(/SUBMIT/i);
  // fireEvent.click(submitButton);
  // screen.debug();
});

})
