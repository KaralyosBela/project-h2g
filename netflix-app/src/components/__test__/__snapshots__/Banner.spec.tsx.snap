// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component should call dispatch after input change with the right action (mocked) 1`] = `
<div>
  <div
    class="picture"
  >
    <div
      class="banner"
    >
      <div
        class="upperBar"
      >
        <div>
          <h4
            class="upperBarTitle"
          >
            <span>
              netflix
            </span>
            Roulette
          </h4>
        </div>
        <button
          class="addButton"
        >
          ADD MOVIE
        </button>
      </div>
      <div
        class="title"
      >
        <h1>
          FIND YOUR MOVIE
        </h1>
      </div>
      <div
        class="searchBar"
      >
        <input
          class="searchInput"
          data-testid="inputField"
          placeholder="What do you want to watch?"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner component should render the component with the 'Add Modal' 1`] = `
<div>
  <div
    class="picture"
  >
    <div>
      Add Movie Modal
    </div>
    <div
      class="banner"
    >
      <div
        class="upperBar"
      >
        <div>
          <h4
            class="upperBarTitle"
          >
            <span>
              netflix
            </span>
            Roulette
          </h4>
        </div>
        <button
          class="addButton"
        >
          ADD MOVIE
        </button>
      </div>
      <div
        class="title"
      >
        <h1>
          FIND YOUR MOVIE
        </h1>
      </div>
      <div
        class="searchBar"
      >
        <input
          class="searchInput"
          data-testid="inputField"
          placeholder="What do you want to watch?"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner component should render the component without the modals 1`] = `
<div>
  <div
    class="picture"
  >
    <div
      class="banner"
    >
      <div
        class="upperBar"
      >
        <div>
          <h4
            class="upperBarTitle"
          >
            <span>
              netflix
            </span>
            Roulette
          </h4>
        </div>
        <button
          class="addButton"
        >
          ADD MOVIE
        </button>
      </div>
      <div
        class="title"
      >
        <h1>
          FIND YOUR MOVIE
        </h1>
      </div>
      <div
        class="searchBar"
      >
        <input
          class="searchInput"
          data-testid="inputField"
          placeholder="What do you want to watch?"
        />
      </div>
    </div>
  </div>
</div>
`;
